src/include/mutex_inline.h

```diff
@@ -313,7 +314,8 @@ __wt_spin_lock_track(WT_SESSION_IMPL *session, WT_SPINLOCK *t)
         else {
             stats[session->stat_bucket][t->stat_app_usecs_off] += (int64_t)time_diff;
         }
-        session_stats[t->stat_session_usecs_off] += (int64_t)time_diff;
+        if (t->stat_session_usecs_off != -1)
```

Comment:
I'm confused by this change. The `stat_int_usecs_off` and `stat_app_usecs_off` values are initialized in the same place as you are now initializing. Why is the code different for this statistic?

I see that this one is a WT_SESSION statistic, and not a WT_CONNECTION statistic, so it doesn't have the additional bucketing scheme as is on the connection stat.

I don't understand why the order of initialization is different. Similarly for updated code below.

Reply:
Currently, in WiredTiger, we track 3 spin locks `schema`, `metadata` and `checkpoint` and only `schema` is followed as part of the session.

As part of the session, we track other statistics along with the schema lock.  Refer `__wt_session_stats`. 
While initializing the `metadata` spinlock, we initialize the `stat_count_off` and etc to a proper offset and leave `stat_session_usecs_off`. Whereas with `schema` spinlock we initialize the `stat_session_usecs_off` too.

During the `__wt_spin_lock_track` of a `metadata` spinlock, where we didn't initialize the `stat_session_usecs_off`, so it points to 0 and the session stats gets incremented whichever is at the 0 offset. In this case, `bytes read` gets incremented for the session stats whenever a `metadata` or `checkpoint` spinlock there is a wait.

To avoid that, we initialize it to -1 and compare it when incrementing.

Reply:
Thanks for explaining. Could you add a comment, something like: "Not all spin locks increment session statistics. Check whether the offset is initialized to determine whether they are enabled."